src/UrlHelper.php

```diff
@@ -157,7 +157,12 @@ private function generateUriFromResult(array $params, RouteResult $result)
 
         $name   = $result->getMatchedRouteName();
         $params = array_merge($result->getMatchedParams(), $params);
-        return $this->router->generateUri($name, $params);
+
+        if ($this->basePath === '/') {
+            return $this->router->generateUri($name, $params);
+        }
+
+        return $this->basePath . $this->router->generateUri($name, $params);
```

Comment:
Hrm... think this is taken care of by `setBasePath()`? I'm pretty much doing exactly what https://github.com/zendframework/zend-expressive-helpers/blob/master/src/UrlHelper.php#L70 does.